WAF - improvements and review (#112)
aziontech/docs_en: web-application-firewall/index.md

[Edit on GitHub <svg width="14" height="14" xmlns="http://www.w3.org/2000/svg"><g fill="none" stroke="#F3652B"><path d="M4.81.71H.672v11.43H12.1V8.001" stroke-width=".8"/><path d="M6.87.786h5.155V5.94M6.31 6.5L12.026.786"/></g></svg>](https://github.com/aziontech/docs_en/edit/master/web-application-firewall/index.md)

> The **Web Application Firewall** (WAF) protects your applications against threats such as SQL Injections, Remote File Inclusion (RFI), Cross-Site Scripting (XSS) and more. WAF scans HTTP and HTTPS requests, detects and blocks malicious activities before they reach your infrastructure and without impacting the performance of your applications.

Reviewer: ```suggestion
The **Web Application Firewall** (WAF) protects your applications against threats such as SQL Injections, Remote File Inclusion (RFI), Cross-Site Scripting (XSS), and more. WAF scans HTTP, and HTTPS requests detect and block malicious activities before reaching your infrastructure and without impacting your applications' performance.
```

Author: Mandou bem @rodrigocst

To avoid blocking licit requests and malfunctioning of your application, you must run a learning stage, in which the WAF Rule Set identifies legitimate behaviors in your application, inserting them into an *allowlist*.

You can also monitor the behavior and effectiveness of your **Web Application Firewall** settings. Through our **Real-Time Events** and **Data Streaming** tools, Azion provides dashboards and reports for online and real-time event log checks. Furthermore, you may import Azion’s log records and manipulate them within your own analysis tools.

> To use your WAF Rule Sets resources you need to enable the Web Application Firewall module in the Edge Firewall Rule Set.

### **Operation Mode**

In order to get maximum performance and accuracy from the product, the learning stage is required. See the two modes of operation to support you in this stage:

| Operation Mode | Description                                                  |
| -------------- | ------------------------------------------------------------ |

> | Counting Mode  | It is used to specify that the WAF should not block any request. Your applications’ traffic will be analyzed and the threats found will only be counted. Use this mode of operation during the first learning stage. |

Reviewer: The squad is adjusting this name from **Counting** to **Learning** and moving it from the WAF Rule Set to the Edge Firewall Rules Engine. It will require changes soon.
@VitorEltz - @wagnerazion

Author: Ok. Let me know as soon I can make these changes, right?

| Attack Family    | The attack family(s) for which the rule increases the score. |

### **Allowlist**

It is the list of legitimate behaviors of your application, which should not increase the *score* of requests. It can be generated automatically during the learning stage or inserted manually through custom rules.

Each blocking rule has *match zones*, as explained in the [Rules](#rules) section. The *Allowlist* aims to disable certain *Match Zones* from a blocking rule.

| Field                | Description                                                  |
| -------------------- | ------------------------------------------------------------ |
| Rule Id              | Unique numeric ID for the blocking rule for which the *allowlist* was generated. |
| Rule Description     | A textual description of what makes the blocking rule for which the *allowlist* was generated. |
| Path                 | When specified, restrict the application of the *allowlist* to the defined *path* only. Path delimits the scope of the *allowlist*. |

> | Whitelist Match Zone | It is the *allowlist* itself. It defines the part or field of the request for which the blocking rule is to be disabled. <br>**Path:** the rule id will not be applied to the request path.<br/>**Query String:** the rule id will not be applied to the query string, also called GET arguments. It can be restricted to both the name and the value of the arguments. It is possible to limit the scope of the *allowlist* to a single GET argument using Conditional Query String.<br/>**Request Header:** the rule id will not be applied to the HTTP headers of the request. It can be restricted to both the name and the value of the headers. It is possible to limit the scope of the *allowlist* to a single HTTP header using Conditional Request Header.<br/>**Request Body:** the rule id will not be applied to the body of a POST, also called POST arguments. It can be restricted to both the name and the value of the arguments. It is possible to limit the scope of the *allowlist* to a single POST argument using Conditional Request Body.<br/>**File Name** (Multipart Body): the rule id will not be applied to the file name in a multipart POST.<br/>**Raw Body:** the rule id will not be applied to the body that was not interpreted from a request, also called an unparsed body. |

Reviewer: ```suggestion
| Allowlist Match Zone | It is the *allowlist* itself. It defines the part or field of the request for which the blocking rule is to be disabled. <br>**Path:** the rule id will not be applied to the request path.<br/>**Query String:** the rule id will not be applied to the query string, also called GET arguments. It can be restricted to both the name and the value of the arguments. It is possible to limit the scope of the *allowlist* to a single GET argument using Conditional Query String.<br/>**Request Header:** the rule id will not be applied to the HTTP headers of the request. It can be restricted to both the name and the value of the headers. It is possible to limit the scope of the *allowlist* to a single HTTP header using the Conditional Request Header.<br/>**Request Body:** the rule id will not be applied to the body of a POST, also called POST arguments. It can be restricted to both the name and the value of the arguments. It is possible to limit the scope of the *allowlist* to a single POST argument using Conditional Request Body.<br/>**File Name** (Multipart Body): the rule id will not be applied to the file name in a multipart POST.<br/>**Raw Body:** the rule id will not be applied to the body that was not interpreted from a request, also called an unparsed body. |
```
